x-pack/plugins/data_enhanced/server/routes/session.ts

```diff
@@ -166,4 +175,43 @@ export function registerSessionRoutes(router: DataEnhancedPluginRouter, logger:
       }
     }
   );
+
+  router.post(
+    {
+      path: '/internal/session/{id}/_extend',
+      validate: {
+        params: schema.object({
+          id: schema.string(),
+        }),
+      },
+    },
+    async (context, request, res) => {
+      const { id } = request.params;
+      try {
+        const config = await config$.pipe(first()).toPromise();
+        const searchSession = (await context.search.getSession(
+          id
+        )) as SavedObject<SearchSessionSavedObjectAttributes>;
+        const expires = moment(searchSession.attributes.expires).add(
```

Comment:
I think the client should pass the final expires attribute, so no calculations are needed.

Reply:
And in general, I think we should have as little logic as possible in the route itself.